src/Middleware/ConcurrencyLimiter/src/QueuePolicies/QueuePolicy.cs

```diff
@@ -10,49 +10,49 @@ namespace Microsoft.AspNetCore.ConcurrencyLimiter
 {
     internal class QueuePolicy : IQueuePolicy, IDisposable
     {
-        private readonly int _maxConcurrentRequests;
-        private readonly int _requestQueueLimit;
+        private readonly int _maxTotalRequest;
         private readonly SemaphoreSlim _serverSemaphore;
 
         private object _totalRequestsLock = new object();
+
+        private readonly static ValueTask<bool> _falseValueTask = new ValueTask<bool>(false);
+
         public int TotalRequests { get; private set; }
 
         public QueuePolicy(IOptions<QueuePolicyOptions> options)
         {
-            _maxConcurrentRequests = options.Value.MaxConcurrentRequests;
-            if (_maxConcurrentRequests <= 0)
+            if (options.Value.MaxConcurrentRequests <= 0)
```

Comment:
nit: I still think you should set `options.Value.MaxConcurrentRequests` in a local variable. Accessing a property multiple times will be slower than a local.

Reply:
Same with `RequestQueueLimit`